modules/python3/processors/pythonscriptsource.cpp

```diff
@@ -0,0 +1,86 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2018 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/python3/processors/pythonscriptsource.h>
+#include <modules/python3/python3module.h>
+#include <inviwo/core/common/inviwoapplication.h>
+#include <inviwo/core/datastructures/geometry/basicmesh.h>
+#include <pybind11/pybind11.h>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo PythonScriptSource::processorInfo_{
+    "org.inviwo.PythonScriptSource",  // Class identifier
+    "Python Script Source",           // Display name
+    "Python",                         // Category
+    CodeState::Experimental,          // Code state
+    "Python, CPU",                    // Tags
+};
+const ProcessorInfo PythonScriptSource::getProcessorInfo() const { return processorInfo_; }
+
+PythonScriptSource::PythonScriptSource()
+    : Processor()
+    , script_("")
+    , scriptFileName_("scriptFileName", "File Name", "")
```

Comment:
Would be great if it is possible to have a default file that shows how it should be used. Possible future PR

Reply:
sure, can do.

Reply:
added example workspace / script for generating a random volume

![image](https://user-images.githubusercontent.com/9251300/38433182-35cc32ec-39ca-11e8-8582-c279ef355769.png)